### PR TITLE
feat(recurrence): Stable link for meeting series

### PR DESCRIPTION
**Pull request by jmtaber129**
# Description

Fixes https://github.com/ParabolInc/parabol/issues/7700

Support stable link meeting redirection for meeting series links of the form `/meeting-series/:meetingId`, e.g. `/meeting-series/kzTbFLeRY4` for series containing a meeting with ID `kzTbFLeRY4`, which will redirect to the most recent meeting in `kzTbFLeRY4`'s series.

If the user does not have access to the meeting (either because it doesn't exist or because it's on a different team), we'll redirect to `/invitation-required`, which will request an invite for the user to the meeting series' team, then redirect once the invite request is accepted.

This also fixes a bug where the `/invitation-required` path won't successfully redirect to a `/meet/:meetingId` route when `meetingId` is inactive (i.e. when the path points to a meeting that already ended).

## Demo
https://www.loom.com/share/bb4ec3c7dad54fcb923b970ab359f2f3

## Testing scenarios
- [ ] Create two users - User A that is on a meeting series' team, and User B that is not.
- [ ] As user A, navigate to `/meeting-series/:meetingId`, where `meetingId` is any meeting ID in the series, and confirm that it redirects to the most recent meeting
  - [ ] Do this when the most recent meeting is active and inactive, and when the `meetingId` is active or inactive.
- [ ] As user B and with user A online, navigate to `/meeting-series/:meetingSeriesId`.
- [ ] Confirm that user B redirects to `/invitation-required` and user A sees an invitation request toast for user B
- [ ] Accept the request as user A
- [ ] Confirm that user B redirects to the most recent meeting in the series

## Final checklist

- [X] I checked the [code review guidelines](../docs/codeReview.md)
- [X] I have added [Metrics Representative](../docs/codeReview.md#metrics-representative) as reviewer(s) if my PR invovles metrics/data/analytics related changes
- [X] I have performed a self-review of my code, the same way I'd do it for any other team member
- [X] I have tested all cases I listed in the testing scenarios and I haven't found any issues or regressions
- [X] Whenever I took a non-obvious choice I added a comment explaining why I did it this way
- [X] I added the label `One Review Required` if the PR introduces only minor changes, does not contain any architectural changes or does not introduce any new patterns and I think one review is sufficient'
- [X] PR title is human readable and could be used in changelog

### CODE REVIEW

**Comment by jmtaber129** [comment]
@BartoszJarocki I'm marking this as "One Review Required" since this doesn't introduce any new patterns and is relatively straightforward, but feel free to loop in maintainers if you think it's worth doing so.

File: packages/client/components/MeetingSeriesRedirector.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      })
+    } else {
+      const {mostRecentMeeting} = meetingSeries
+      const {id: meetingId} = mostRecentMeeting
+
+      history.replace({
+        pathname: `/meet/${meetingId}`
+      })
+    }
+  }, [])
```
Comment (BartoszJarocki) [inline packages/client/components/MeetingSeriesRedirector.tsx:51]: +1 as this is a one time thing, there's no need to use effect here. you can simply use react router [redirect component](https://v5.reactrouter.com/web/api/Redirect). declarative and cleaner

File: packages/client/components/MeetingSeriesRoot.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+const MeetingRoot = () => {
+  const {history, match} = useRouter<{meetingSeriesId: string}>()
+  const {params} = match
+  const {meetingSeriesId: encodedMeetingSeriesId} = params
+  const meetingSeriesId = decodeURIComponent(encodedMeetingSeriesId)
+  useEffect(() => {
+    if (!meetingSeriesId) {
+      history.replace('/meetings')
+    }
+  }, [])
```
Comment (BartoszJarocki) [inline packages/client/components/MeetingSeriesRoot.tsx:18]: +1 same here, better to use `<Redirect />`, although it'll have to be below usage of `useQueryLoaderNow` but that's ok

File: packages/server/graphql/types/User.ts [hunk 3]
```diff
@@ -525,17 +526,22 @@ const User: GraphQLObjectType<any, GQLContext> = new GraphQLObjectType<any, GQLC
           description:
             'The meetingId to check for the invitation, if teamId not available (e.g. on a meeting route)'
         },
+        meetingSeriesId: {
+          type: GraphQLID,
+          description:
+            'The meetingSeriesId to check for the invitation, if teamId not available (e.g. on a meeting series route)'
+        },
```
Comment (BartoszJarocki) [inline packages/server/graphql/types/User.ts:533]: +1 where do we use it? do you mind sharing a little bit more what's the use case here? having troubles understanding what's the problem we're solving 🙏

**Review by BartoszJarocki — COMMENTED** [review]
A couple of comments, will do the testing and final round after they're resolved. nice work!

File: packages/client/components/MeetingSeriesRedirector.tsx (hunk 1 above)
Comment (jmtaber129) [reply packages/client/components/MeetingSeriesRedirector.tsx]: TIL about the `Redirect` component!

**Review by BartoszJarocki — COMMENTED** [review]
@jmtaber129 I tested and it's all good. the only worry I have right now is we shouldn't be using postgres int ids in links. I believe link should look like `meeting-series/kxvmKg4St2` instead `meeting-series/meetingSeries:120`. The reasons: is easy to figure out meeting ids outside of their teams for our users, it's also easy to see how often this feature is used. Do you have any thoughts about that?

**Comment by jmtaber129** [comment]
@BartoszJarocki hmm yea you're right.  Not sure what alternative approaches there are though 🙁  meetings look like `/meet/kxvmKg4St2` largely because rethink IDs look like `kxvmKg4St2`.

**Comment by BartoszJarocki** [comment]
@jmtaber129 probably a wild idea but what if...
1. ID of the first meeting in the meeting series would be used in the url `meeting-series/:firstMeetingId`
2. Then the query in `MeetingSeriesRedirector` would be something like 
```
  const data = usePreloadedQuery<MeetingSeriesRedirectorQuery>(
    graphql`
      query MeetingSeriesRedirectorQuery($meetingId: ID!) {
        viewer {
          isConnected
          meeting(meetingId: $meetingId) {
            meetingSeries {
               mostRecentMeeting {
                  id
               }
            }
          }
        }
      }
    `,
    queryRef,
    {UNSTABLE_renderPolicy: 'full'}
  )
```
and then... everything else stays the same? am I missing something or would that work just fine?

**Comment by jmtaber129** [comment]
@BartoszJarocki I actually really like that! So `/meeting-series/:meetingId` would redirect to the most recent meeting in the meeting's series.

I'll change this to that behavior.

**Comment by BartoszJarocki** [comment]
awesome!

File: packages/server/graphql/public/mutations/acceptTeamInvitation.ts [hunk 4]
```diff
@@ -51,8 +51,6 @@ const acceptTeamInvitation: MutationResolvers['acceptTeamInvitation'] = async (
   const {invitation} = invitationRes
   const {meetingId, teamId, invitedBy: inviterId} = invitation
   const acceptAt = invitation.meetingId ? 'meeting' : 'team'
-  const meeting = meetingId ? await dataLoader.get('newMeetings').load(meetingId) : null
-  const activeMeetingId = meeting && !meeting.endedAt ? meetingId : null
```
Comment (jmtaber129) [inline packages/server/graphql/public/mutations/acceptTeamInvitation.ts:55]: @mattkrick I'd like to make this change as part of a fix for `/invitation-required` not redirecting when `meetingId` points to a meeting that has already ended - are there any negative side effects I might not be seeing here?

**Review by BartoszJarocki — APPROVED** [review]
lgtm!

**Pull request merged**